src/smaug/errors.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+export function TokenError(message) {
+  this.name = 'TokenError';
+  this.message = message || 'Generic TokenError';
+  this.stack = (new Error()).stack;
+  this.httpStatusCode = 400;
+  this.httpError = 'generic_token_error';
+}
+TokenError.prototype = Object.create(Error.prototype);
+TokenError.prototype.constructor = TokenError;
```

Comment:
> I am not sure why you set constructor manually, - as far as I remember it should be set automatically when you make a new instance. Looks good otherwise :)

I just did whatever MDN suggested: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error